src/main/java/io/strimzi/kafka/access/Utils.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.access;
+
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.OwnerReference;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.javaoperatorsdk.operator.processing.event.ResourceID;
+import io.strimzi.api.kafka.model.Kafka;
+import io.strimzi.api.kafka.model.KafkaBuilder;
+import io.strimzi.kafka.access.model.KafkaAccess;
+import io.strimzi.kafka.access.model.KafkaAccessSpec;
+import io.strimzi.kafka.access.model.KafkaReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Utils {
+
+    public static final String MANAGED_BY_LABEL_KEY = "app.kubernetes.io/managed-by";
+    public static final String INSTANCE_LABEL_KEY = "app.kubernetes.io/instance";
+    public static final String STRIMZI_CLUSTER_LABEL_VALUE = "strimzi-cluster-operator";
+    public static final String KAFKA_ACCESS_LABEL_VALUE = "kafka-access-operator";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Utils.class);
+
+    /**
+     * Filters the stream of KafkaAccess objects to find only the ones that reference the provided Kafka resource.
+     *
+     * @param kafkaAccessList    Stream of KafkaAccess objects in the current cache
+     * @param kafka              Kafka resource to check for in the KafkaAccess objects
+     *
+     * @return                   Set of ResourceIDs for the KafkaAccess objects that reference the Kafka resource
+     */
+    public static Set<ResourceID> getKafkaAccessResourceIDsForKafkaInstance(final Stream<KafkaAccess> kafkaAccessList, final Kafka kafka) {
+        final Optional<String> kafkaName = Optional.ofNullable(kafka.getMetadata()).map(ObjectMeta::getName);
+        final Optional<String> kafkaNamespace = Optional.ofNullable(kafka.getMetadata()).map(ObjectMeta::getNamespace);
+        if (kafkaName.isEmpty() || kafkaNamespace.isEmpty()) {
+            LOGGER.error("getKafkaAccessResourceIDsForKafkaInstance called with Kafka that is missing metadata, returning empty set");
+            return Collections.emptySet();
+        }
+        return kafkaAccessList
+                .filter(kafkaAccess -> Optional.ofNullable(kafkaAccess.getSpec())
+                        .map(KafkaAccessSpec::getKafka)
+                        .map(KafkaReference::getName)
+                        .isPresent())
```

Comment:
This is a good point, I don't think the currently generated CRD yaml file has any sections required, I'll investigate how that works with java-operator-sdk/fabric8. Do we perhaps leave it for now and remove it when we add in the required fields?

Reply:
AFAIK, adding this annotation on fields should do the trick

`javax.validation.constraints.NotNull;`

The Java Operator SDK should generated the CRD putting the fields in the `required` section. 
Let me know if that works otherwise we can move forward with the current code and open an issue to track this.